.github/workflows/tests.yml

```
@@ -0,0 +1,57 @@
+name: 'Test Suites'
+
+on: [pull_request, push]
+
+jobs:
+  tests:
+    name: 'test suites'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: "${{ matrix.shell }} {0}"
+
+    strategy:
+      matrix:
```

```suggestion
    strategy:
      fail-fast: false
      matrix:
```
this way one failure won't stop the rest from finishing